query-engine/core/src/interpreter/query_interpreters/read.rs

```diff
@@ -107,96 +107,123 @@ fn read_related<'a, 'b>(
             }
         };
 
-        let other_fields: Vec<_> = query
-            .parent_field
-            .related_field()
-            .linking_fields()
-            .fields()
-            .flat_map(|f| f.data_source_fields())
-            .collect();
-
-        let filters: Vec<Filter> = relation_parent_ids
-            .clone()
-            .into_iter()
-            .map(|id| {
-                let filters = id
-                    .pairs
-                    .into_iter()
-                    .zip(other_fields.iter())
-                    .map(|((_, value), other_field)| other_field.equals(value))
-                    .collect();
-                Filter::and(filters)
-            })
-            .collect();
-
-        let filter = Filter::or(filters);
-        let mut args = query.args.clone();
-
-        args.filter = match args.filter {
-            Some(existing_filter) => Some(Filter::and(vec![existing_filter, filter])),
-            None => Some(filter),
+        let relation = query.parent_field.relation();
+
+
+
+        dbg!("RELATION: {}", &query.parent_field.relation().manifestation);
+
+        dbg!("READ RELATED", &parent_result);
+
+        // prisma level join does not work for many 2 many yet
+        // can only work if we have a parent result. This is not the case when we e.g. have nested delete inside an update
+        let use_prisma_level_join = !relation.is_many_to_many() && parent_result.is_some();
```

Comment:
on master this snippet is even more strict:
`let use_prisma_level_join = !relation.is_many_to_many() && parent_result.is_some() && !query.args.is_with_pagination();`